app/javascript/components/supplemental_application/SupplementalApplicationContainer.js

```diff
@@ -142,99 +147,81 @@ const SupplementalApplicationContainer = ({ store }) => {
-            onSubmit={(submittedValues) =>
-              handleStatusModalSubmit(
-                submittedValues,
-                convertPercentAndCurrency(form.getState().values)
-              )
-            }
-          />
-        </>
+    <>
+      {failed && (
+        <AlertBox
+          invert
+          onCloseClick={() => setFailed(false)}
+          message='Please resolve any errors before saving the application.'
+        />
       )}
-    />
+      <form onSubmit={handleSubmit} style={{ margin: '0px' }} id='shortForm' noValidate>
+        <AsymColumnLayout.Container>
+          <AsymColumnLayout.MainContent>
+            <ConfirmedPreferencesSection
+              application={state.application}
+              applicationMembers={getApplicationMembers(state.application)}
+              fileBaseUrl={state.fileBaseUrl}
+              onSave={(preferenceIndex, formApplicationValues) =>
+                updateSavedPreference(dispatch, preferenceIndex, formApplicationValues)
+              }
+              onDismissError={() => preferencesFailedChanged(dispatch, false)}
+              confirmedPreferencesFailed={state.confirmedPreferencesFailed}
+              form={form}
+            />
+            <Income listingAmiCharts={state.listingAmiCharts} visited={visited} form={form} />
+            <LeaseSection
+              form={form}
+              values={values}
+              showLeaseSection={hasLease(state.leaseSectionState)}
+              onCreateLeaseClick={() => leaseCreated(dispatch)}
+            />
+            <DemographicsSection />
+          </AsymColumnLayout.MainContent>
+          <AsymColumnLayout.Sidebar>
+            <Sidebar
+              statusHistory={state.statusHistory}
+              loading={state.loading}
+              onAddCommentClicked={() => handleAddCommentClicked(form, touched)}
+              onChangeStatus={(value) => onChangeStatus(form, touched, value)}
+              onSaveClicked={() => checkForValidationErrors(form, touched)}
+            />
+          </AsymColumnLayout.Sidebar>
+        </AsymColumnLayout.Container>
+      </form>
+      <StatusModalWrapper
+        alertMsg={state.statusModal.alertMsg}
+        isOpen={state.statusModal.isOpen}
+        loading={state.statusModal.loading}
+        onAlertCloseClick={() => closeSuppAppStatusModalAlert(dispatch)}
+        onClose={() => closeSuppAppStatusModal(dispatch)}
+        onSubmit={(submittedValues) => {
+          const { application: prevApplication, leaseSectionState } = state
+          return submitSuppAppStatusModal(
+            dispatch,
+            submittedValues,
+            convertPercentAndCurrency(form.getState().values),
+            prevApplication,
+            leaseSectionState
+          )
+        }}
+        showAlert={state.statusModal.showAlert}
+        status={state.statusModal.status}
+        submitButton={state.statusModal.isInAddCommentMode ? 'Save' : 'Update'}
```

Comment:
Instead of setting this text in the statusModal state object, I switched this so the state object has the more "state-y" field isInAddCommentMode, and the render function can decide what text to show.